frontends/common/options.cpp

```diff
@@ -130,6 +133,38 @@ void CompilerOptions::setInputFile() {
     }
 }
 
+std::vector<const char*>* CompilerOptions::process(int argc, char* const argv[]) {
+    char buffer[PATH_MAX];
+    int len;
+    struct stat st;
+    if ((len = readlink("/proc/self/exe", buffer, sizeof(buffer))) > 0 ||
```

Comment:
Is this stuff portable on other operating systems?
Maybe it should be in a Linux-specific library. We haven't been too careful about this kind of system dependences so far.